content/blog/data-layer-performant-editing.md

```diff
@@ -0,0 +1,72 @@
+
+```jsx
+"scripts": {
+    "dev": "yarn tinacms server:start -c \"next dev\"",
+    "build": "yarn tinacms server:start -c \"next build\"",
+    "start": "yarn tinacms server:start -c \"next start\" --experimentalData",
+  },
+```
+
+Once the flag is added and the CLI has been run, Tina will update the generated schema which will let Tina know you want to use the Data Layer. 
+
+> Once you have generated the Schema, you will need to commit the changes to GitHub for it to start working on the project.
+
+## What does the Data Layer do to increase Performance?
+
+Once the Data Layer is enabled on your project, we will automatically synchronize a copy of your repository with our secure cloud database. After Tina has run the initial indexing of your repository Tina will automatically index new or updated content. This is done behind the scenes and you won’t notice that we are doing this, except the increase in performance when editing.
+
+### When we might do a full re-index
+
+In some cases we might have to re-index your project, this is usually happens when the core of the project has changed for example:
+
+ - Changes to `.tina\schema.ts`
+
+ - Changes to the path to [.tina](https://tina.io/docs/tina-cloud/faq/#does-tina-cloud-work-with-monorepos)
+
+ - Changes to the configured [repository](https://tina.io/docs/tina-cloud/dashboard/projects/#changing-the-repository)
+
+ - New branches in GitHub
+
+### Some things to note
+
+This is still an experimental feature and the following things should be considered before enabling on your project: 
+
+1\. The index process isn’t exposed to the end user, which means it is possible that in an event a partially indexed site could return incomplete results
+
+2\. GitHub has a API request limitation of 5000 requests per repository per hour, if you have an extremely large project you could hit this rate limit. If you have a repository with more than 1500 items please do not activate this feature.
+
+## The future
+
+The team at Tina has plans for the following highly requested features as the Data Layer matures out of its experimental phase. In fact we have already begun working on some of the features mentioned below, so stay tuned! 
+
+### More complex and advanced queries
+
+The Data Layer opens up our GraphQL layer to be even more powerful, in the future Tina plans to offer: 
+
+1. Pagination
+
+2. Filtering
+
+3. Sorting 
+
+This means you will be able to reduce some of your calls if you don’t need a full data set, a good example of this would be where you only need 3 blog posts for a features section. In the current Tina integration you will need to retrieve all of the posts and filter them down after the fact, in the future you won’t need to do this. 
```

Comment:
```suggestion
This means you will be able to reduce some of your calls if you don’t need a full data set. A good example of this would be where you only need 3 blog posts for a features section. In the current Tina integration, you need to retrieve all of the posts and filter them down after the fact. In the future you won’t need to do this. 
```